src/core/lib/security/credentials/tls/grpc_tls_certificate_provider.cc

```diff
@@ -198,13 +198,38 @@ void FileWatcherCertificateProvider::ForceUpdate() {
       root_certificate_ = "";
     }
   }
+  absl::StatusOr<bool> keyCertMatch = false;
+  //check whether the present credentials are valid.
+  if (!pem_key_cert_pairs_.empty()) {
+    for (int i = 0; i < pem_key_cert_pairs_.size(); i++) {
+      keyCertMatch =
+          PrivateKeyAndCertificateMatch(pem_key_cert_pairs_.at(i).private_key(),
+                                        pem_key_cert_pairs_.at(i).cert_chain());
+      if (!(keyCertMatch.ok() && *keyCertMatch)) {
+        break;
+      }
+    }
+  }
   const bool identity_cert_changed =
       (!pem_key_cert_pairs.has_value() && !pem_key_cert_pairs_.empty()) ||
       (pem_key_cert_pairs.has_value() &&
        pem_key_cert_pairs_ != *pem_key_cert_pairs);
   if (identity_cert_changed) {
     if (pem_key_cert_pairs.has_value()) {
-      pem_key_cert_pairs_ = std::move(*pem_key_cert_pairs);
+      for (int i = 0; i < pem_key_cert_pairs->size(); i++) {
+        keyCertMatch = PrivateKeyAndCertificateMatch(
+          pem_key_cert_pairs->at(i).private_key(), pem_key_cert_pairs->at(i).cert_chain());
+        if (!(keyCertMatch.ok() && *keyCertMatch)) {
+          gpr_log(GPR_ERROR,
```

Comment:
Why are you logging the match results? Is this for debugging purposes?

Reply:
Mostly, yes

Reply:
SG. Do keep in mind you'll have to remove it at the end.

Reply:
The information would also be helpful when *gRPC users* are debugging certain types of errors. For example, if this fails, the users might get very confused on what's going wrong, since the most obvious result is that the connection is dropped.
Having this logged will help them identify the problem. They could open up the logs gRPC created, and see those lines. They would probably notice something wrong with their keys, and go ahead updating them.
So let's have the logging back.